### PR TITLE
Add drive command suppliers

### LINKED ISSUE #30: Maybe move drive commands to start using Suppliers?
There's potentially a better way to supply axis values to drive commands than our current method of having static methods to get them in RobotContainer, which is exemplified in [YAGSL's example commands](https://github.com/BroncBotz3481/YAGSL-Example/blob/main/src/main/java/frc/robot/commands/swervedrive/drivebase/AbsoluteDrive.java).

This will make the commands more flexible, which is cool.

### PR DESCRIPTION
**Pull request by QuackitsQuinn**
closes #30

### CODE REVIEW

File: src/main/java/frc/robot/commands/drive/DriveFieldOriented.java [hunk 1]
```diff
@@ -16,13 +16,21 @@
 import frc.robot.Constants;
 import frc.robot.RobotContainer;
 import frc.robot.subsystems.Drive;
+import java.util.function.Supplier;
 
 /** Drives the robot in field-oriented mode. */
 public class DriveFieldOriented extends Command {
   private final Drive drive;
+  private final Supplier<Double> xScaledSupplier, yScaledSupplier, zScaledSupplier;
```
Comment (willitcode) [inline src/main/java/frc/robot/commands/drive/DriveFieldOriented.java:24]: Rename these to xSupplier, ySupplier, etc. In this context we don't care that the controller axes are scaled.

File: src/main/java/frc/robot/commands/drive/DriveFieldOrientedHeadingSnapping.java [hunk 2]
```diff
@@ -27,11 +28,27 @@
 public class DriveFieldOrientedHeadingSnapping extends Command {
   private Drive drive;
   private XboxController controller;
+  private final Supplier<Double> xScaledSupplier, yScaledSupplier, zScaledSupplier;
```
Comment (willitcode) [inline src/main/java/frc/robot/commands/drive/DriveFieldOrientedHeadingSnapping.java:31]: Rename these to xSupplier, ySupplier, etc. In this context we don't care that the controller axes are scaled.

File: src/main/java/frc/robot/commands/drive/DriveFieldOrientedHeadingSnapping.java [hunk 3]
```diff
@@ -46,33 +63,26 @@ public void initialize() {
...
 
     Translation2d translation = SwerveController.getTranslation2d(desiredSpeeds);
-    if (headingX == 0
-        && headingY == 0
-        && Math.abs(RobotContainer.getScaledControllerRightXAxis()) > 0) {
+    if (headingX == 0 && headingY == 0 && Math.abs(zScaledSupplier.get()) > 0) {
       drive.driveFieldOriented(
-          translation,
-          (RobotContainer.getScaledControllerRightXAxis() * Constants.Drive.MAX_ANGULAR_SPEED));
+          translation, (zScaledSupplier.get()) * Constants.Drive.MAX_ANGULAR_SPEED);
```
Comment (willitcode) [inline src/main/java/frc/robot/commands/drive/DriveFieldOrientedHeadingSnapping.java:85]: why is `zScaledSupplier.get()` wrapped in parentheses?

File: src/main/java/frc/robot/commands/drive/DriveRobotOriented.java [hunk 4]
```diff
@@ -16,14 +16,22 @@
 import frc.robot.Constants;
 import frc.robot.RobotContainer;
 import frc.robot.subsystems.Drive;
+import java.util.function.Supplier;
 
 /** Drives the robot in robot-oriented mode. Default command for {@link Drive} subsystem. */
 public class DriveRobotOriented extends Command {
   private final Drive drive;
+  private final Supplier<Double> xScaledSupplier, yScaledSupplier, zScaledSupplier;
```
Comment (willitcode) [inline src/main/java/frc/robot/commands/drive/DriveRobotOriented.java:24]: Rename these to xSupplier, ySupplier, etc. In this context we don't care that the controller axes are scaled.

**Review by willitcode — CHANGES_REQUESTED** [review]
JavaDoc all of the constructors for the commands please

File: src/main/java/frc/robot/commands/drive/DriveFieldOrientedHeadingSnapping.java [hunk 5]
```diff
@@ -27,11 +28,38 @@
...
+   * Drives the robot field oriented with heading snapping. All values are suppliers to make the
+   * command more versatile.
+   *
+   * @param xSupplier The x speed
+   * @param ySupplier The y speed
+   * @param zSupplier The z speed
+   * @param upSupplier If the robot should go up
+   * @param downSupplier If the robot should go down
+   * @param leftSupplier If the robot should go left
+   * @param rightSupplier If the robot should go right
```
Comment (willitcode) [inline src/main/java/frc/robot/commands/drive/DriveFieldOrientedHeadingSnapping.java:44]: "If true, the robot heading will be set to move towards (the opposing alliance wall, our alliance wall, left, or right)"

**Review by willitcode — CHANGES_REQUESTED** [review]
Clarify that the X, Y, and Z suppliers should all be joystick values on the interval [-1, 1] and that they're back-left positive. (They're definitely not speeds).

File: src/main/java/frc/robot/RobotContainer.java [hunk 6]
```diff
@@ -74,14 +76,38 @@ public class RobotContainer {
...
+  private Supplier<Double> scaledControllerLeftXAxisSupplier = () -> getScaledControllerLeftXAxis();
+  private Supplier<Double> scaledControllerLeftYAxisSupplier = () -> getScaledControllerLeftYAxis();
+  private Supplier<Double> scaledControllerRightXAxisSupplier =
+      () -> getScaledControllerRightXAxis();
+  private Supplier<Double> scaledControllerRightYAxisSupplier =
+      () -> getScaledControllerRightYAxis();
+  private Supplier<Boolean> povUpDirectionSupplier = () -> rawXboxController.getPOV() == 0;
+  private Supplier<Boolean> povRightDirectionSupplier = () -> rawXboxController.getPOV() == 90;
+  private Supplier<Boolean> povDownDirectionSupplier = () -> rawXboxController.getPOV() == 180;
+  private Supplier<Boolean> povLeftDirectionSupplier = () -> rawXboxController.getPOV() == 270;
```
Comment (willitcode) [inline src/main/java/frc/robot/RobotContainer.java:93]: Maybe we move these down into `configureBindings()`? idk, thoughts?

Comment (willitcode) [reply src/main/java/frc/robot/RobotContainer.java:93]: Just the POV suppliers, not the axis suppliers

Comment (QuackitsQuinn) [reply src/main/java/frc/robot/RobotContainer.java:93]: I think thats a good idea

Comment (QuackitsQuinn) [reply src/main/java/frc/robot/RobotContainer.java:93]: im a liar, dont do that

**Pull request merged**